Source/JavaScriptCore/runtime/StringPrototypeInlines.h

```diff
@@ -196,6 +196,29 @@ ALWAYS_INLINE JSString* jsSpliceSubstringsWithSeparators(JSGlobalObject* globalO
 }
 
 enum class StringReplaceSubstitutions : bool { No, Yes };
+template<StringReplaceSubstitutions substitutions>
+ALWAYS_INLINE String tryMakeReplacedString(const String& string, const String& replacement, size_t matchStart, size_t matchEnd)
+{
+    String result;
+    if constexpr (substitutions == StringReplaceSubstitutions::Yes) {
+        size_t dollarPos = replacement.find('$');
+        if (dollarPos != WTF::notFound) {
+            StringBuilder builder(OverflowPolicy::RecordOverflow);
+            int ovector[2] = { static_cast<int>(matchStart), static_cast<int>(matchEnd) };
+            substituteBackreferencesSlow(builder, replacement, string, ovector, nullptr, dollarPos);
+            if (UNLIKELY(builder.hasOverflowed()))
+                return { };
+            result = tryMakeString(StringView(string).substring(0, matchStart), builder.toString(), StringView(string).substring(matchEnd, string.length() - matchEnd));
```

Comment:
How about just return here when `result` is non null?
```
auto result = tryMakeString(StringView(string).substring(0, matchStart), builder.toString(), StringView(string).substring(matchEnd, string.length() - matchEnd));
if (result)
    return result;
```

Reply:
Fixed!